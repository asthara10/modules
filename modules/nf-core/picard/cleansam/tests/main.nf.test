nextflow_process {

    name "Test Process PICARD_CLEANSAM"
    script "../main.nf"
    process "PICARD_CLEANSAM"

    tag "modules"
    tag "modules_nfcore"
    tag "picard"
    tag "picard/cleansam"

    test("test_picard_cleansam") {


        when {
            process {
                """
                
                input[0] = [ 
        [ id:'test', single_end:true ], // meta map
        file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true) 
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

}
