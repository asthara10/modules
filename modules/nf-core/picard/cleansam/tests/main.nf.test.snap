{
    "versions": {
        "content": [
            [
                "versions.yml:md5,d59ef6e139d691a25979a8a4ed9e5a5b"
            ]
        ],
        "timestamp": "2023-11-24T13:09:37.056692"
    }
}